compute_sdk/globus_compute_sdk/sdk/batch.py

```diff
@@ -2,13 +2,29 @@
 
 import typing as t
 from collections import defaultdict
+from dataclasses import asdict, dataclass
 
 from globus_compute_sdk.sdk.utils.uuid_like import UUID_LIKE_T
 from globus_compute_sdk.serialize import ComputeSerializer
 
 _default_serde = ComputeSerializer()
 
 
+@dataclass
+class UserRuntime:
+    """Information about a user's runtime environment, which is sent along with task
+    submissions to MEP user config templates.
```

Comment:
Being precise, it's not sent to the config _template_, but as context variable to the template renderer.  (Not a blocker; your call how or if you want to address this.)